rest/model/host.go

```diff
@@ -311,23 +310,7 @@ type APIOffboardUserResults struct {
 	TerminatedVolumes []string `json:"terminated_volumes"`
 }
 
-// APIHostProvisioningOptions represents script to provision a host that
-// is meant to be executed in a particular environment type.
+// APIHostProvisioningOptions represents script to provision a host.
 type APIHostProvisioningOptions struct {
-	Directive string `json:"directive"`
-	Content   string `json:"content"`
-}
-
-// BuildFromService converts from service level structs to an APIHost. It can
-// be called multiple times with different data types, a service layer host and
-// a service layer task, which are each loaded into the data structure.
-func (a *APIHostProvisioningOptions) BuildFromService(i interface{}) error {
-	switch opts := i.(type) {
-	case *userdata.Options:
-		a.Directive = string(opts.Directive)
-		a.Content = opts.Content
-		return nil
-	default:
-		return errors.Errorf("invalid type %T, expected user data options", i)
-	}
+	Content string `json:"content"`
```

Comment:
I guess we could change it to just be a string and not return a JSON response, but I think Evergreen's REST routes should send/receive JSON instead of raw strings and it'll be easier to change in the future if we need to add more options.

Reply:
that makes sense to me too.
Now that it's just a script, though, should we name the thing `APIHostProvisioningScript`?

Reply:
I think it's fine as is. The route is `/rest/v1/host/{host_id}/provisioning_options` and I don't want to assume it'll always be one long shell script.